### PR TITLE
Chore: Onboard @carbon/react to IBM Telemetry 🚀

### PR DESCRIPTION
**Pull request by francinelucca**
- Uninstall @carbon/telemetry from @carbon/react
- Add `telemetry.yml` config file
- Update postinstall script to call new telemetry script, remove legacy script call
- Add IBM Telemetry collection disclosure to README

### CODE REVIEW

**Review by tay1orjones — APPROVED** [review]
🎉 🙌 🚀

**Comment by tay1orjones** [comment]
I'd like to get this same change made to the v10 branch. We can do it, or you, just let me know.

File: packages/react/telemetry.yml [hunk 1]
```diff
@@ -0,0 +1,815 @@
+# yaml-language-server: $schema=https://unpkg.com/@ibm/telemetry-config-schema@v0/dist/config.schema.json
```
Comment (mattrosno) [inline packages/react/telemetry.yml]: This config was generated with a script by @francinelucca to get all prop and prop values that we want to collect. We're still finishing the script to use when instrumenting other packages. Once that script is done, _maybe_ we could add that to a CI/CD script to automate this config's upkeep as components, component props, and supported component prop values change.

**Comment by mattrosno** [comment]
> I'd like to get this same change made to the v10 branch. We can do it, or you, just let me know.

@tay1orjones let's instrument v11 first, then we can look at the v10 branch. We'll need to generate the allowed telemetry prop config for the v10 branch.

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**